### PR TITLE
Allow setting timestamp of metrics.

### PR DESCRIPTION
**Pull request by Grundlefleck**
*Issue #, if available:* 61
https://github.com/awslabs/aws-embedded-metrics-node/issues/61

*Description of changes:*
Adds the MetricsLogger.setTimestamp() method.

One aspect I think core maintainers will have a much better sense of, if I set the timestamp should that be preserved across calls to `flush()`? I wasn't 100% sure what would be desired behaviour, so I took a guess and went for "yes, if set explicitly, preserve". For my own use case described in #61, I will be calling `setTimeout` in 100% of `MetricsContext` instances generated, so it would be irrelevant anyway.

Another guess I made was that if the timestamp is not set explicitly, the `MetricsContext` should reset the `meta.Timestamp` field to `new Date()`. I think this is less surprising behaviour for users who never call `setTimestamp`. Plus if I preserved the timestamp when not explicitly set, that would change the behaviour for all existing users, I thought it would only make sense to change behaviour for callers who are "opting in" by using `setTimeout`. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jaredcnance** [comment]
Thanks for the PR! I think this is the correct behavior. Preserving timestamp across flushes makes sense.

File: README.md [hunk 1]
```diff
@@ -191,9 +191,25 @@ Examples:
 setNamespace("MyApplication");
 ```
 
+- **setTimestamp**(Date | number timestamp)
+
+Sets the CloudWatch [timestamp](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#about_timestamp) that extracted metrics are associated with. If not set a default value of `new Date()` will be used.
+
+If set, timestamp will be preserved across calls to flush().
```
Comment (jaredcnance) [inline README.md:198]: ...for a given `MetricsLogger` instance.

File: README.md [hunk 2]
```diff
@@ -191,9 +191,25 @@ Examples:
...
+
+Requirements: 
+ * Date or Unix epoch millis, up to two weeks in the past and up to two hours in the future, as enforced by [CloudWatch](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#about_timestamp). 
+
+Examples:
+
+```py
```
Comment (jaredcnance) [inline README.md:205]: js

**Review by jaredcnance — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Grundlefleck) [reply README.md]: Have updated, also removed the single other use of ```py that lead to my copypasta fail.

File: README.md (hunk 1 above)
Comment (Grundlefleck) [reply README.md]: Clarified in a subsequent commit.

**Comment by Grundlefleck** [comment]
@jaredcnance I can't see the CodeBuild failures but given 
  a) tests pass locally* and 
  b) CodeBuild passed on a previous commit and the only changes where to README.md

Does the build maybe need a retry? If you find it is something with the code please let me know and I'll try to reproduce and address it.

\* I know, I know, "works on my machine" :laughing:

**Comment by jaredcnance** [comment]
Build was failing because we are getting throttled by Docker Hub when pulling the CW Agent. We need to switch to using ECR, I believe. I rebuilt it and it passed, but I'm not sure why the status check hasn't been reported back to GitHub. I'm approving and merging the PR. Thanks!

**Pull request merged**